src/main/java/net/coderbot/iris/uniforms/PropertiesUniforms.java

```diff
@@ -0,0 +1,82 @@
+package net.coderbot.iris.uniforms;
+
+import net.coderbot.iris.Iris;
+import net.coderbot.iris.gl.program.ProgramBuilder;
+import net.coderbot.iris.gl.uniform.UniformUpdateFrequency;
+import net.coderbot.iris.shaderpack.IdMapParser;
+import net.minecraft.block.Block;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.render.block.entity.BlockEntityRenderDispatcher;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.Hand;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import java.util.Map;
+import java.util.function.IntSupplier;
+
+public final class PropertiesUniforms {
```

Comment:
I don't like the name of this class, it's a bit ambiguous since it's possible to define custom uniforms in `shaders.properties` as well.